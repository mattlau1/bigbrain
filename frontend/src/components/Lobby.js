import React, { useState } from 'react'
import { Col, Container, Image, Row } from 'react-bootstrap'
import { useParams } from 'react-router';
import styled from 'styled-components';
import ReactPlayer from 'react-player';
import RangeSlider from 'react-bootstrap-range-slider';
import cat from '../assets/cat.gif'

const LobbyContainer = styled(Container)`
  background-color: #44A3E5;
  height: 100vh;
  color: white;
  padding: 0;
`

const LobbyHeading = styled.h2`
  font-family: 'Montserrat', sans-serif;
  font-size: 72pt;
  color: #333333;
  margin-bottom: 0;
  font-weight: 1000;

  @media (max-width: 992px) {
    font-size: 56pt;
  }
`

const LobbyMediumText = styled(LobbyHeading)`
  font-size: 56pt;
  @media (max-width: 992px) {
    font-size: 36pt;
  }
`

const LobbySmallText = styled(LobbyHeading)`
  font-size: 24pt;
  @media (max-width: 992px) {
    font-size: 20pt;
  }
`

const Lobby = () => {
  const { sessionId } = useParams();

  // default volume
  const [volume, setVolume] = useState(10);

  // background music playlist
  const songs = [
    'https://www.youtube.com/watch?v=u1HyZ2hmxnc',
    'https://www.youtube.com/watch?v=HbbGCld47mA'
  ]

  return (
    <LobbyContainer className="p-2" fluid>
      <ReactPlayer url={songs}
        playing={true}
        controls={false}
        loop={true}
        height={0}
        volume={volume / 100}
      />
      <Container className="pb-4" style={{ backgroundColor: '#5AADE7' }} fluid>
        <Row md={8} className="d-flex justify-content-center pt-4 mb-0 px-4 mx-4">
          <LobbySmallText className="p-2 bg-light overflow-visible">Lobby ID</LobbySmallText>
        </Row>
        <Row md={8} className="d-flex justify-content-center">
          <LobbyHeading className="p-4 text-center bg-light overflow-visible">{sessionId}</LobbyHeading>
        </Row>

      </Container>
      <Container>
        <Col>
          <Row md={12} className="d-flex justify-content-center pt-4 mb-4 px-4 mx-4">
            <LobbyMediumText className="p-4 mx-4 text-center bg-light overflow-visible">
              Waiting ...
            </LobbyMediumText>
          </Row>
          <Row className="d-flex justify-content-center" md={4}>
            <Col className="px-0" md={4}>
              <Row md={12} className="d-flex justify-content-center pt-4 mb-0">
                <h4 className="text-center">Volume</h4>
              </Row>
              <Row md={12} className="d-flex justify-content-center pt-4 mb-0 px-4">
                <RangeSlider
                  id="volume-slider"
                  value={volume}
                  onChange={e => setVolume(e.target.value)}
                  max={100}
                  min={0}
                  step={1}
                  tooltip={'auto'}
                />
              </Row>
            </Col>
          </Row>
        </Col>
      </Container>
      <div className="fixed-bottom">
        <Image src={cat}></Image>
      </div>
      <div className="fixed-bottom d-flex justify-content-end">
        <Image style={{ transform: 'scaleX(-1)' }} src={cat}></Image>
      </div>
    </LobbyContainer>
  )
}

export default Lobby
