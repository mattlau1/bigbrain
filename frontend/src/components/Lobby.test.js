import { shallow } from 'enzyme';
import React from 'react';
import Lobby from './Lobby';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Lobby', () => {
  it('should not render volume control', () => {
    const wrapper = shallow(<Router><Lobby /></Router>);
    expect(wrapper.find('#volume-slider')).toHaveLength(0);
  })

  it('should render the lobby page', () => {
    expect(shallow(<Router><Lobby /></Router>)).toEqual({});
  })
})
