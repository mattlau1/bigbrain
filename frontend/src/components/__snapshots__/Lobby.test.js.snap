// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lobby should not render volume control 1`] = `
<div
  className="sc-bdvvaa eSxlee p-2 container-fluid"
>
  <div
    style={
      Object {
        "height": 0,
        "width": "640px",
      }
    }
  />
  <div
    className="pb-4 container-fluid"
    style={
      Object {
        "backgroundColor": "#5AADE7",
      }
    }
  >
    <div
      className="d-flex justify-content-center pt-4 mb-0 px-4 mx-4 row row-cols-md-8"
    >
      <h2
        className="sc-gsDJrp sc-hKwCoD ccXyhz lkUUbE p-2 bg-light overflow-visible"
      >
        Lobby ID
      </h2>
    </div>
    <div
      className="d-flex justify-content-center row row-cols-md-8"
    >
      <h2
        className="sc-gsDJrp ccXyhz p-4 text-center bg-light overflow-visible"
      />
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="col"
    >
      <div
        className="d-flex justify-content-center pt-4 mb-4 px-4 mx-4 row row-cols-md-12"
      >
        <h2
          className="sc-gsDJrp sc-dkPtyc ccXyhz kvDAwL p-4 mx-4 text-center bg-light overflow-visible"
        >
          Waiting ...
        </h2>
      </div>
      <div
        className="d-flex justify-content-center row row-cols-md-4"
      >
        <div
          className="px-0 col-md-4"
        >
          <div
            className="d-flex justify-content-center pt-4 mb-0 row row-cols-md-12"
          >
            <h4
              className="text-center"
            >
              Volume
            </h4>
          </div>
          <div
            className="d-flex justify-content-center pt-4 mb-0 px-4 row row-cols-md-12"
          >
            <span
              className="range-slider__wrap"
            >
              <input
                className="range-slider range-slider--primary"
                disabled={false}
                max={100}
                min={0}
                onChange={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                step={1}
                type="range"
                value={10}
              />
              <div
                className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
                style={
                  Object {
                    "left": "calc(10% + 8px)",
                  }
                }
              >
                <div
                  className="range-slider__tooltip__label"
                >
                  10
                </div>
                <div
                  className="range-slider__tooltip__arrow"
                />
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="fixed-bottom row"
  >
    <img
      className="d-none d-md-block d-lg-block d-xl-block"
      src="cat.gif"
    />
  </div>
  <div
    className="fixed-bottom d-flex justify-content-end  row"
  >
    <img
      className="d-none d-md-block d-lg-block d-xl-block"
      src="cat.gif"
      style={
        Object {
          "transform": "scaleX(-1)",
        }
      }
    />
  </div>
</div>
`;
